File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    q = deque(sorted(people, reverse = True))
    
    answer = 0
    while q:
        big = q.popleft()
        
        if q and big + q[-1] <= limit:
            q.pop()
        answer += 1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        int f = 0;
        int b = people.length - 1;
        
        Arrays.sort(people);
        
        while (f <= b) {
            answer++;
            if (people[f] + people[b] <= limit) {
                f++;
            }
            b--;
        }
        
        return answer;
    }
}